packages/gatsby/src/db/nodes.js

```diff
@@ -23,31 +23,32 @@ interface NodeStore {
 /**
  * Get content for a node from the plugin that created it.
  *
- * @param {Object} node
- * @returns {promise}
+ * @param {IGatsbyNode} node
+ * @returns {Promise<string>}
  */
-function loadNodeContent(node) {
+async function loadNodeContent(node) {
   if (_.isString(node.internal.content)) {
     return Promise.resolve(node.internal.content)
-  } else {
-    return new Promise(resolve => {
-      // Load plugin's loader function
-      const plugin = store
-        .getState()
-        .flattenedPlugins.find(plug => plug.name === node.internal.owner)
-      const { loadNodeContent } = require(plugin.resolve)
-      if (!loadNodeContent) {
-        throw new Error(
-          `Could not find function loadNodeContent for plugin ${plugin.name}`
-        )
-      }
-
-      return loadNodeContent(node).then(content => {
-        // TODO update node's content field here.
-        resolve(content)
-      })
-    })
   }
+
+  // Load plugin's loader function
+  const plugin = store
+    .getState()
+    .flattenedPlugins.find(plug => plug.name === node.internal.owner)
+
+  const { loadNodeContent } = require(plugin.resolve)
+
+  if (!loadNodeContent) {
+    throw new Error(
+      `Could not find function loadNodeContent for plugin ${plugin.name}`
+    )
+  }
+
+  const content = loadNodeContent(node)
+
+  node.internal.context = content
```

Comment:
This is new, right?

Basically resolving this TODO?

> // TODO update node's content field here.

Reply:
You're right. I didn't intend to do this here actually. But I guess we might as well. I'll update the PR title/desc.